File: dataset/human/python/file_275.py
object Main {

  def main(args: Array[String]): Unit = {
    val sc = new java.util.Scanner(System.in)

    val N = sc.nextInt()
    val A = Array.fill(N)(sc.nextInt())
    val isEven = A.count(_ < 0) % 2 == 0

    val absSum = A.map(a => Math.abs(a)).foldLeft(0L)(_ + _)

    val res =
      if (isEven) absSum
      else absSum - (2 * A.map(a => Math.abs(a)).min)
    println(res)
  }

}

File: dataset/human/java/file_275.java
import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.LongStream;

public class Main {

    public static void main(String[] args) {

        execute();
    }

    private static void execute() {

        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();

        long[] A = new long[N];
        long[] AbsM = new long[N];
        for (int i = 0; i < N; i++) {
            A[i] = sc.nextLong();
            AbsM[i] = Math.abs(A[i]);
        }

        long negativeCount = LongStream.of(A).filter(s -> s < 0).count();
        long absAmount = LongStream.of(AbsM).reduce(0, (s1, s2) -> s1 + s2);

        if (negativeCount % 2 == 0) {
            System.out.println(absAmount);
        } else {
            Arrays.sort(AbsM);
            System.out.println(absAmount - AbsM[0] * 2);
        }
    }

}
